packages/core/src/indexing/postgres/postgres-index-api.ts

```diff
@@ -63,14 +64,62 @@ export class PostgresIndexApi implements DatabaseIndexApi {
   }
 
   async indexModels(models: Array<IndexModelArgs>): Promise<void> {
+    if (models.length === 0) return
+
     await initMidTables(this.dbConnection, models, this.logger)
     await this.verifyTables(models)
+    // FIXME: populate the updated_by field properly when auth is implemented
+    await this.dbConnection(INDEXED_MODEL_CONFIG_TABLE_NAME)
+      .insert(models.map(indexModelArgs => {
+        return {
+          model: indexModelArgs.model.toString(),
+          updated_by: "<FIXME: PUT ADMIN DID WHEN AUTH IS IMPLEMENTED>"
+        }
+      }))
+      .onConflict('model')
+      .merge({
+        updated_at: this.dbConnection.fn.now(),
+        is_indexed: true,
+        updated_by: "<FIXME: PUT ADMIN DID WHEN AUTH IS IMPLEMENTED>"
+      })
+
     const modelStreamIDs = models.map((args) => args.model)
     this.modelsToIndex.push(...modelStreamIDs)
   }
 
+  async stopIndexingModels(models: Array<StreamID>): Promise<void> {
+    if (models.length === 0) return
+
+    // FIXME: populate the updated_by field properly when auth is implemented
+    await this.dbConnection(INDEXED_MODEL_CONFIG_TABLE_NAME)
+      .insert(models.map(model => {
+        return {
+          model: model.toString(),
+          is_indexed: false,
+          updated_by: "<FIXME: PUT ADMIN DID WHEN AUTH IS IMPLEMENTED>"
+        }
+      }))
+      .onConflict('model')
+      
+      .merge({
+        updated_at: this.dbConnection.fn.now(),
+        is_indexed: false,
+        updated_by: "<FIXME: PUT ADMIN DID WHEN AUTH IS IMPLEMENTED>"
+      })
+    const modelsAsStrings = models.map(streamID => streamID.toString())
+    this.modelsToIndex = this.modelsToIndex.filter(modelStreamID => !modelsAsStrings.includes(modelStreamID.toString()))
+  }
+
   async init(): Promise<void> {
     await initConfigTables(this.dbConnection, this.logger)
+    this.modelsToIndex = (await this.dbConnection(INDEXED_MODEL_CONFIG_TABLE_NAME)
```

Comment:
helper function please!  I think all things that do SQL should be in separate functions so we don't have sql interspersing random other higher-level code.  Having a line like `this.modelsToIndex = await this.getIndexedModelsFromDatabase()` will be much clearer to read IMHO